hoot-core/src/main/cpp/hoot/core/conflate/DiffConflator.cpp

```diff
@@ -177,28 +178,10 @@ MemChangesetProviderPtr DiffConflator::getTagDiff()
   return _pTagChanges;
 }
 
-void DiffConflator::storeOriginalIDs(OsmMapPtr& pMap)
+void DiffConflator::storeOriginalMap(OsmMapPtr& pMap)
 {
-  // Nodes
-  const NodeMap &nodes = pMap->getNodes();
-  for (HashMap<long, NodePtr>::const_iterator it = nodes.begin(); it != nodes.end(); ++it)
-  {
-    _originalIds.insert(ElementId(ElementType::Node, it->first));
-  }
-
-  // Ways
-  const WayMap &ways = pMap->getWays();
-  for (HashMap<long, WayPtr>::const_iterator it = ways.begin(); it != ways.end(); ++it)
-  {
-    _originalIds.insert(ElementId(ElementType::Way, it->first));
-  }
-
-  // Relations
-  const RelationMap &relations = pMap->getRelations();
-  for (HashMap<long, RelationPtr>::const_iterator it = relations.begin(); it != relations.end(); ++it)
-  {
-    _originalIds.insert(ElementId(ElementType::Relation, it->first));
-  }
+  // Use the copy constructor
```

Comment:
My only suggestion might be a quick check if any element is set as `Unknown2` since this should only contain a map of `Unknown1` elements.  Then log an error or warning.  If the this copy has both `Unknown1` and `Unknown2` then will tag differential still work?  Should you continue on or stop the process?

Reply:
That's a good suggestion... a little bullet-proofing here could go a long way.
If the map contained uknown1 & unknown2 elements, the tag-diff would not deliver expected results.

Reply:
Well done.  Approve-ified.